packages/jsii/lib/assembler.ts

```diff
@@ -384,27 +390,35 @@ export class Assembler implements Emitter {
                 this._diagnostic(clause, ts.DiagnosticCategory.Error, `Ignoring ${ts.SyntaxKind[clause.token]} heritage clause`);
                 continue;
             }
+
             for (const expression of clause.types) {
                 const iface = this._typeChecker.getTypeFromTypeNode(expression);
+
+                // if this interface is internal/private, we erase it from the API definition
+                if (this._isPrivateOrInternal(iface.symbol)) {
+                    LOG.trace(`erasing hidden interface ${iface.symbol.name}`);
+                    continue;
```

Comment:
When the interface is `@internal`, then you want to check if that interface has non-`@internal` super-interfaces and register those instead (because interface implementation/extension is transitive). This is also true for `_visitInterface`.

Reply:
Good point. Shoot.